Kudu.Core/Jobs/BaseJobRunner.cs

```diff
@@ -72,17 +71,65 @@ protected string WorkingDirectory
 
         protected JobSettings JobSettings { get; set; }
 
-        private static int CalculateHashForJob(string jobBinariesPath)
+        internal static bool JobDirectoryHasChanged(
```

Comment:
One it has changed, might be nice to log the file that caused it. Would have saved me a lot of time if we had had that when I first debugged the .exe.config condition :)